stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -1008,24 +979,51 @@ public func _tffunc<In : TensorGroup, Out : TensorGroup>(
   return traceContext.specializeTFFunction(with: [])
 }
 
-/// Returns a valid TF device string such as
-/// "/job:localhost/replica:0/task:0/device:CPU:0", which corresponds to the
-/// closest enclosing withDevice() construct.
-/// A return value of nil indicates the absence of withDevice().
 internal extension _ExecutionContext {
-  @usableFromInline
+  /// Returns a valid TensorFlow device name such as, which corresponds to the
+  /// closest enclosing withDevice() call.
+  /// A return value of nil indicates the absence of withDevice() or the
+  /// immediately enclosing withDefaultDevice() call.
   var currentDeviceName: String? {
-    if let (kind, index) = _ThreadLocalState.value._currentDevice {
-      switch kind {
-      case .cpu:
-        return "\(cpuDeviceNamePrefix)\(index)"
-      case .gpu:
-        return "\(gpuDeviceNamePrefix!)\(index)"
-      case .tpu:
-        return "\(tpuDeviceNamePrefix!)\(index)"
-      }
+    return deviceScopes.last ?? nil
+  }
+
+  /// See documentation for the top-level `withDevice` function.
+  func withDevice<R>(_ kind: DeviceKind, _ index: UInt = 0,
+                     perform body: () throws -> R) rethrows -> R {
+    var name: String
```

Comment:
Since all assignments to `name` are initialization (not mutation), this variable can be a `let`.

```suggestion
    let name: String
```